sdk/keyvault/azure-security-keyvault-certificates/inc/azure/keyvault/certificates/certificate_client_models.hpp

```diff
@@ -0,0 +1,772 @@
+#include <azure/core/http/http.hpp>
+#include <azure/core/nullable.hpp>
+#include <azure/core/response.hpp>
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace Azure { namespace Security { namespace KeyVault { namespace Certificates {
+
+  /**
+   * @brief Contains identity and other basic properties of a Certificate.
+   *
+   */
+  struct CertificateProperties final
+  {
+    // Attributes
+
+    /**
+     * @brief Indicate when the certificate will be valid and can be used for cryptographic
+     * operations.
+     *
+     */
+    Azure::Nullable<Azure::DateTime> NotBefore;
+
+    /**
+     * @brief Indicate when the certificate will expire and cannot be used for cryptographic
+     * operations.
+     *
+     */
+    Azure::Nullable<Azure::DateTime> ExpiresOn;
+
+    /**
+     * @brief Indicate when the certificate was created.
+     *
+     */
+    Azure::Nullable<Azure::DateTime> CreatedOn;
+
+    /**
+     * @brief Indicate when the certificate was updated.
+     *
+     */
+    Azure::Nullable<Azure::DateTime> UpdatedOn;
+
+    /**
+     * @brief The number of days a certificate is retained before being deleted for a soft
+     * delete-enabled Key Vault.
+     *
+     */
+    Azure::Nullable<int> RecoverableDays;
+
+    /**
+     * @brief The recovery level currently in effect for keys in the Key Vault.
+     *
+     * @remark If Purgeable, the certificate can be permanently deleted by an authorized user;
+     * otherwise, only the service can purge the keys at the end of the retention interval.
+     *
+     */
+    Azure::Nullable<std::string> RecoveryLevel;
+
+    // Properties
+
+    /**
+     * @brief Dictionary of tags with specific metadata about the certificate.
+     *
+     */
+    std::unordered_map<std::string, std::string> Tags;
+
+    /**
+     * @brief The name of the certificate.
+     *
+     */
+    std::string Name;
+
+    /**
+     * @brief The certificate identifier.
+     *
+     */
+    std::string Id;
+
+    /**
+     * @brief The Key Vault base Url.
+     *
+     */
+    std::string VaultUrl;
+
+    /**
+     * @brief The version of the certificate.
+     *
+     */
+    std::string Version;
+
+    /**
+     * @brief Get the digital thumbprint of the certificate which can be used to uniquely identify
+     * it.
+     *
+     */
+    std::vector<uint8_t> X509Thumbprint;
+
+    /**
+     * @brief Indicate whether the certificate is enabled and useable for cryptographic operations.
+     *
+     */
+    Azure::Nullable<bool> Enabled;
+
+    /**
+     * @brief Construct a new Certificate Properties object
+     *
+     */
+    CertificateProperties() = default;
+
+    /**
+     * @brief Construct a new Certificate Properties object
+     *
+     * @param name The name of the certificate.
+     */
+    CertificateProperties(std::string const& name) : Name(name) {}
+  };
+
+  /**
+   * @brief An Azure Key Vault certificate.
+   *
+   */
+  struct KeyVaultCertificate
```

Comment:
This struct is not final and is being inherited, so add a virtual destructor:
`virtual ~KeyVaultCertificate() = default;`